### PR TITLE
Add runtime id for Arch Linux

### PR DESCRIPTION
**Pull request by omajid**
Like gentoo, Arch Linux is a rolling release and doesn't have explicit version numbers.

```
$ docker run -it archlinux/base cat /etc/os-release
NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="0;36"
HOME_URL="https://www.archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://bugs.archlinux.org/"
LOGO=archlinux
```

cc @alucryd

### CODE REVIEW

File: pkg/Microsoft.NETCore.Platforms/runtime.compatibility.json
Comment (lpereira) [inline pkg/Microsoft.NETCore.Platforms/runtime.compatibility.json]: I'd call this `archlinux` just so it's not ambiguous to people that don't know it's a Linux distro.  Otherwise, seeing things like `arch-x64` will be pretty confusing (are we talking about "x64 architecture"?)

Comment (omajid) [reply pkg/Microsoft.NETCore.Platforms/runtime.compatibility.json]: I see what you are saying, but that is going to be a muchmore involved fix. As it is, ID from /etc/os-release says `arch` so most RID computation in dotnet computes that to `arch-x64`. Fixing everything that computes the RID is going to be tricky and special-casing Arch Linux. Should I go down that route?

Comment (ViktorHofer) [reply pkg/Microsoft.NETCore.Platforms/runtime.compatibility.json]: @ericstj suggestions?

Comment (ericstj) [reply pkg/Microsoft.NETCore.Platforms/runtime.compatibility.json]: Interesting, this is the first time we've looked at a linux distro name that collides with another concept in the RID graph (let's hope someone doesn't come up with Arm linux).  

RIDs aren't exactly supposed to be user friendly, so I don't think its a blocker, but I do agree this is confusing.  WWAPMD? (what would a PM do?)  @terrajobst @richlander

**Comment by bartonjs** [comment]
Syntactically it seems fine, I could go either way on special-casing the RID resolver (archlinux is a nicer identifier, IMO, but special-casing things is less nice)... but I will ask my standard RID questions of "do we need this?" and "is this better served from a RID-extension package?"

As far as I know, we're not directly supporting Arch, so it seems better served by some mythical graph-extension package.  If build from source is far enough along that they want to adopt us, and this is the only real hiccup... then let's do this and pop some champagne :champagne: (whether or not one drinks, the sound definitely is satisfying).

**Comment by alucryd** [comment]
@bartonjs Arch Linux maintainer of the dotnet core packages here. I've built dotnet from source (almost) since my initial push to our official repositories, so as far as adoption goes, I'm all in. The source build only broke with the 3.0 release, as long as it's fixed rapidly (preferably before arch users burn my house for not updating to 3.0 quickly enough), I'm completely fine with how you name and where you get this RID from :)

**Comment by maryamariyan** [comment]
Thank you for your contribution. As announced in dotnet/coreclr#27549 this repository will be moving to dotnet/runtime on November 13. If you would like to continue working on this PR after this date, the easiest way to move the change to dotnet/runtime is:
1.	In your corefx repository clone, create patch by running `git format-patch origin`
2.	In your runtime repository clone, apply the patch by running `git apply --directory src/corefx <path to the patch created in step 1>`

**Comment by bartonjs** [comment]
@alucryd What RID is used on your current Arch build? (I presume `arch-x64`)  And how is it resolving via the graph? (I presume you have a distro patch similar to this PR)

**Comment by fmauNeko** [comment]
@bartonjs You can find the build files there: https://git.archlinux.org/svntogit/community.git/tree/trunk?h=packages/dotnet-core

**Comment by bartonjs** [comment]
Neat.  https://git.archlinux.org/svntogit/community.git/plain/trunk/dotnet-coreclr-rid.patch?h=packages/dotnet-core does seem like `arch-x64` is the currently used RID on arch, but I don't see where the `arch-x64 : arch, linux-x64`, `arch : linux` parts of the RID graph come into play... I'm a little surprised that package restore isn't upset.

Since `arch-x64` is what Arch is already using, and `arch-x64` is what the tooling is producing automatically, and there's a package/ebuild/whatever-arch-calls-it for us already... I think there's sufficient reason to take this PR.

@omajid Can you rebase/merge to pick up the other RID graph changes? I'm cool with merging once the merge button is OK with it.

**Comment by omajid** [comment]
@bartonjs does this look good to go now?

**Review by bartonjs — APPROVED** [review]

**Pull request merged**
